homeassistant/components/wink/climate.py

```diff
@@ -234,7 +236,14 @@ def hvac_mode(self) -> str:
             return HVAC_MODE_HEAT
         if wink_mode == "eco":
             return HVAC_MODE_AUTO
-        return WINK_HVAC_TO_HA.get(wink_mode)
+
+        # This "if" should never happen, so this is just to
```

Comment:
Why not just:

`return WINK_HVAC_TO_HA.get(wink_mode, HVAC_MODE_AUTO)`

Anyways, this is a functional change, actually, the above `eco` mode that is assigned to `HVAC_MODE_AUTO` is already possibly incorrect. `HVAC_MODE_AUTO` is meant for schedules or AI/Learned behavior.

Saying it's auto, while it is unknown, is maybe wrong/odd as well. Should it maybe just blow up in that case instead?

Reply:
I agree that HVAC_MODE_AUTO is sort of the "poor mans choice", I tried a while ago to HVAC_MODE_UNKNOWN accepted, but without any luck. I have seen a number of integration the return HVAC_MODE_AUTO always, as to say "we do not know what it is".

Made your suggestion in the 2 places it happens.

Reply:
Unresolving the conversation.

I'm not convinced just returning `HVAC_MODE_AUTO` is correct or justified. When would this case happen for Wink?

Reply:
Honestly, I think this should not occur even.

`return WINK_HVAC_TO_HA[wink_mode]`

Reply:
Alternative action taken.